cmd/whitesourceExecuteScan.go

```diff
@@ -657,39 +659,43 @@ func readAssessmentsFromFile(assessmentFilePath string, utils whitesourceUtils)
 	exists, err := utils.FileExists(assessmentFilePath)
 	if err != nil {
 		log.SetErrorCategory(log.ErrorConfiguration)
-		log.Entry().Errorf("unable to check existence of assessment file at '%s'", assessmentFilePath)
+		log.Entry().WithError(err).Errorf("unable to check existence of assessment file at '%s'", assessmentFilePath)
 	}
 	assessmentFile, err := utils.Open(assessmentFilePath)
 	if exists && err != nil {
 		log.SetErrorCategory(log.ErrorConfiguration)
-		log.Entry().Errorf("unable to open assessment file at '%s'", assessmentFilePath)
+		log.Entry().WithError(err).Errorf("unable to open assessment file at '%s'", assessmentFilePath)
 	}
 	assessments := &[]format.Assessment{}
 	if exists {
 		defer assessmentFile.Close()
 		assessments, err = format.ReadAssessments(assessmentFile)
 		if err != nil {
 			log.SetErrorCategory(log.ErrorConfiguration)
-			log.Entry().Errorf("unable to parse assessment file at '%s'", assessmentFilePath)
+			log.Entry().WithError(err).Errorf("unable to parse assessment file at '%s'", assessmentFilePath)
 		}
 	}
 	return assessments
 }
 
 // checkSecurityViolations checks security violations and returns an error if the configured severity limit is crossed.
-func checkProjectSecurityViolations(config *ScanOptions, cvssSeverityLimit float64, project ws.Project, sys whitesource, assessments *[]format.Assessment, influx *whitesourceExecuteScanInflux) (int, []ws.Alert, error) {
+func checkProjectSecurityViolations(config *ScanOptions, cvssSeverityLimit float64, project ws.Project, sys whitesource, assessments *[]format.Assessment, influx *whitesourceExecuteScanInflux) (int, []ws.Alert, []ws.Alert, error) {
 	// get project alerts (vulnerabilities)
+	assessedAlerts := []ws.Alert{}
 	alerts, err := sys.GetProjectAlertsByType(project.Token, "SECURITY_VULNERABILITY")
 	if err != nil {
-		return 0, alerts, fmt.Errorf("failed to retrieve project alerts from WhiteSource: %w", err)
+		return 0, alerts, assessedAlerts, fmt.Errorf("failed to retrieve project alerts from WhiteSource: %w", err)
 	}
 
 	// filter alerts related to existing assessments
 	filteredAlerts := []ws.Alert{}
-	if len(*assessments) > 0 {
+	if assessments != nil && len(*assessments) > 0 {
 		for _, alert := range alerts {
 			if result, err := alert.ContainedIn(assessments); err == nil && result == false {
 				filteredAlerts = append(filteredAlerts, alert)
+			} else if alert.Assessment != nil {
```

Comment:
isn't the `else if` also executed when `alert.ContainedIn(...` results in an error? Is this really intended or doesn't this harm due to the check on the assessment?

Reply:
That does not harm due to the check on `Assessment` which will only be set once the assessments where successfully read, interpreted and matched to an `Alert`.